### PR TITLE
fix File::create() permission

### PR DESCRIPTION
**Pull request by TaikiKawamura**
Document (File API : http://book.cakephp.org/2.0/en/core-utility-libraries/file-folder.html)
says "Create a new file with 0644 permissions".
but, $mode doesn't affect created file.

### CODE REVIEW

File: lib/Cake/Utility/File.php [hunk 1]
```diff
@@ -105,10 +105,11 @@ public function __destruct() {
  * @return bool Success
  * @link http://book.cakephp.org/2.0/en/core-utility-libraries/file-folder.html#File::create
  */
-	public function create() {
+	public function create($mode = 0755) {
```
Comment (markstory) [inline lib/Cake/Utility/File.php]: I'm not sure this is a safe value that we want to promote more of.

File: lib/Cake/Utility/File.php [hunk 2]
```diff
@@ -89,7 +89,7 @@ public function __construct($path, $create = false, $mode = 0755) {
 			$this->name = basename($path);
 		}
 		$this->pwd();
-		$create && !$this->exists() && $this->safe($path) && $this->create();
+		$create && !$this->exists() && $this->safe($path) && $this->create($mode);
```
Comment (AD7six) [inline lib/Cake/Utility/File.php]: The `$mode` there is supposed to be (awful api...) [the mode of the _containing folder](https://github.com/cakephp/cakephp/blob/master/lib/Cake/Utility/File.php#L83).

**Comment by AD7six** [comment]
Hmz whether the code needs to change, the [docs are wrong](http://book.cakephp.org/2.0/en/core-utility-libraries/file-folder.html#file-api) since the mode paramter isn't supposed to apply to the file. Creating directories without execute perms would more or less make the dir more or less inaccessible.

File: lib/Cake/Utility/File.php (hunk 2 above)
Comment (ADmad) [inline lib/Cake/Utility/File.php]: The `$create` param too is used for the directory as well as file. I agree awfullness of this API. We should at least make it sensible in 3.0.

Comment (ADmad) [inline lib/Cake/Utility/File.php]: Creating a file should not be creating parent directory too imho.

**Pull request closed without merging**